wrapper/src/main/java/software/amazon/jdbc/states/SessionStateServiceImpl.java

```diff
@@ -78,6 +82,20 @@ public void setupPristineAutoCommit() throws SQLException {
       return;
     }
     this.sessionState.autoCommit.setPristineValue(this.pluginService.getCurrentConnection().getAutoCommit());
+    this.logCurrentState();
+  }
+
+  @Override
+  public void setupPristineAutoCommit(final boolean autoCommit) throws SQLException {
```

Comment:
can you make this parameter optional and then only have one setupPristineAutoCommit() function instead of two that do almost the same thing except in one line?

Reply:
I suspect that it's be hard to distinguish the case when the value is explicitly provided from the case when the value is not provided and default value is used. For example, for `setupPristingCatalog(String catalog)`, `null `is a legit value when it's provided.